pss/pss.go

```diff
@@ -144,681 +120,32 @@ type Pss struct {
-// described below. The recipient address can be of any length, and the byte slice will be matched
-// to the MSB slice of the peer address of the equivalent length.
-//
-// If the recipient address (or partial address) is within the neighbourhood depth of the forwarding
-// node, then it will be forwarded to all the nearest neighbours of the forwarding node. In case of
-// partial address, it should be forwarded to all the peers matching the partial address, if there
-// are any; otherwise only to one peer, closest to the recipient address. In any case, if the message
-//// forwarding fails, the node should try to forward it to the next best peer, until the message is
-//// successfully forwarded to at least one peer.
-func (p *Pss) forward(msg *message.Message) error {
-	defer metrics.GetOrRegisterResettingTimer("pss/forward", nil).UpdateSince(time.Now())
-	sent := 0 // number of successful sends
-	to := make([]byte, addressLength)
-	copy(to[:len(msg.To)], msg.To)
-	neighbourhoodDepth := p.NeighbourhoodDepth()
-
-	// luminosity is the opposite of darkness. the more bytes are removed from the address, the higher is darkness,
-	// but the luminosity is less. here luminosity equals the number of bits given in the destination address.
-	luminosityRadius := len(msg.To) * 8
-
-	// proximity order function matching up to neighbourhoodDepth bits (po <= neighbourhoodDepth)
-	pof := pot.DefaultPof(neighbourhoodDepth)
-
-	// soft threshold for msg broadcast
-	broadcastThreshold, _ := pof(to, p.BaseAddr(), 0)
-	if broadcastThreshold > luminosityRadius {
-		broadcastThreshold = luminosityRadius
-	}
-
-	var onlySendOnce bool // indicates if the message should only be sent to one peer with closest address
-
-	// if measured from the recipient address as opposed to the base address (see Kademlia.EachConn
-	// call below), then peers that fall in the same proximity bin as recipient address will appear
-	// [at least] one bit closer, but only if these additional bits are given in the recipient address.
-	if broadcastThreshold < luminosityRadius && broadcastThreshold < neighbourhoodDepth {
-		broadcastThreshold++
-		onlySendOnce = true
+		return nil, err
 	}
 
-	p.kademliaLB.EachBinDesc(to, func(bin network.LBBin) bool {
-		if bin.ProximityOrder < broadcastThreshold && sent > 0 {
-			// This bin is at the same distance as the node to the message. If already sent, we stop sending
-			return false
-		}
-		for _, lbPeer := range bin.LBPeers {
-			if sendFunc(p, lbPeer.Peer, msg) {
-				lbPeer.AddUseCount()
-				sent++
-				if onlySendOnce {
-					return false
-				}
-				if bin.ProximityOrder == addressLength*8 {
-					// stop iterating if successfully sent to the exact recipient (perfect match of full address)
-					return false //stop iterating
-				}
-			}
-		}
-		return true
-	})
-
-	// cache the message
-	p.addFwdCache(msg)
+	//TODO: verify correctness of tc?, that it will hit it's targets, should this be in the trojan package?
```

Comment:
there is no need, `Put` already calls a validator on the submitted chunk

Reply:
and if you mean the correctness of the chunk as a trojan chunk: then yes, this is covered in the `trojan` package.

regardless, as far as i understand it, we don't have a way of actually knowing for sure if a chunk ever gets to its destination

(but this is way above what we're dealing with here, i think)

Reply:
I removed this comment, as the correctness is covered in the `trojan` package.
This was a question I had, and you resolved with this comment, so thanks.